### PR TITLE
fix(windows): Welcome should always show in front

### LINKED ISSUE #4549: bug(windows): the welcome page gets hidden being the Configuration after the installation is done
**Describe the bug**

After completing the installation of the keyboard, a welcome page is not shown. It seems like it gets buried under the Configuration window.

**To Reproduce**

1. Go to 'Configuration'
2. Click on 'Download keyboard...'
3. Choose a keyboard to install and finish the process

**Expected behavior**

The welcome page has usually been shown on top of the Configuration. 

**Screenshots**

image

---

**Keyman for Windows:**
 - OS: Windows 10 Pro 20H2 x64
 - Keyman version: 14.0.248-beta
 - Target Application: Configuration

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #4549.

The Package Welcome dialog would not always show in front because its owner could be `nil` in some circumstances.

### CODE REVIEW

**Comment by MakaraSok** [comment]
[The build](https://build.palaso.org/repository/download/KeymanDesktop_TestPullRequests/260526:id/release/14.0.258/keyman-14.0.258.exe) was tested and proved to not fix. The same behavior is encountered -- when installed from "Install keyboard...," the welcome is hidden behind the Configuration.

image

**Review by mcdurdin — COMMENTED** [review]
Just checking

**Review by mcdurdin — COMMENTED** [review]
More testing

**Comment by MakaraSok** [comment]
The build is not accessible at the moment: 
image

**Comment by mcdurdin** [comment]
> The build is not accessible at the moment:

Yes, I haven't done any work on this, so we will get a new build when we look at the issue again

**Comment by MakaraSok** [comment]
Retested and confirm that the welcome page shows on top of the configuration after the installation processes have completed.

image

File: windows/src/global/delphi/general/KeymanPaths.pas [hunk 1]
```diff
@@ -373,6 +375,24 @@ class function TKeymanPaths.KeymanConfigStaticHttpFilesPath(const filename: stri
   Result := Result + filename;
 end;
 
+class function TKeymanPaths.KeymanCustomisationPath: string;
+var
```
Comment (jahorton) [inline windows/src/global/delphi/general/KeymanPaths.pas:379]: As this isn't mentioned in the PR's description... it looks like the changes in this function also "throw in" a fix for ensuring the 'branded' version of the welcome page is presented correctly if one exists?

That's obviously somewhat related, but it's definitely a separate detail from the PR's title.

**Review by rc-swag — APPROVED** [review]
lgtm

File: windows/src/global/delphi/general/KeymanPaths.pas (hunk 1 above)
Comment (mcdurdin) [reply windows/src/global/delphi/general/KeymanPaths.pas:379]: Yeah, I should have mentioned this in the description sorry. This makes the search for the 'customisation' file (which is a really old term...) work properly for development machines as well as for installable builds. It's also consistent with other path searches.

**Review by jahorton — APPROVED** [review]

File: windows/src/desktop/kmshell/main/initprog.pas [hunk 2]
```diff
@@ -614,7 +614,7 @@ procedure ShowKeyboardWelcome(PackageName: WideString);  // I2569
 begin
   n := kmcom.Packages.IndexOf(PackageName);
   if n >= 0 then
-    DoShowPackageWelcome(kmcom.Packages[n], True);
+    DoShowPackageWelcome(nil, kmcom.Packages[n], True);
```
Comment (ermshiperete) [inline windows/src/desktop/kmshell/main/initprog.pas:617]: Should we pass `Screen.ActiveForm` instead of `nil`?

Comment (mcdurdin) [reply windows/src/desktop/kmshell/main/initprog.pas:617]: In this code path, `Screen.ActiveForm` will always be `nil` anyway.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.96-alpha](https://keyman.com/downloads/releases/alpha/15.0.96)

**Pull request merged**